File: DXDAL/ProductionSchedule.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Namespace SPGData
    Public Class ProductionSchedule
        Inherits AuditableBase

        Private _finish As DateTime
        Private _start As DateTime
        Private _machineLine As MachineLine
        Private _people As Double?
        Private _machineOperator As Double?
        Private _productionItem As Items
        Private _mustRun As Boolean?
        Private _priorityLevel As Integer?
        Private _project As Project
        Private _projectDetail As ProjectDetails
        Private _scheduleNote As String
        Private _Quantity As Double
        Private _scheduleByQuantity As Boolean

        Public Property Finish() As DateTime
            Get
                Return _finish
            End Get
            Set(value As DateTime)
                SetPropertyValue(Of DateTime)("Finish", _finish, value)
            End Set
        End Property

        Public Property Start() As DateTime
            Get
                Return _start
            End Get
            Set(value As DateTime)
                SetPropertyValue(Of DateTime)("Start", _start, value)
            End Set
        End Property

        Public Property MachineLine() As MachineLine
            Get
                Return _machineLine
            End Get
            Set(value As MachineLine)
                SetPropertyValue(Of MachineLine)("MachineLine", _machineLine, value)
            End Set
        End Property

        Public Property People As Double?
            Get
                Return _people
            End Get
            Set(value As Double?)
                SetPropertyValue(Of Double?)("People", _people, value)
            End Set
        End Property

        Public Property MachineOperator As Double?
            Get
                Return _machineOperator
            End Get
            Set(value As Double?)
                SetPropertyValue(Of Double?)("MachineOperator", _machineOperator, value)
            End Set
        End Property

        Public Property ProductionItem As Items
            Get
                Return _productionItem
            End Get
            Set(value As Items)
                SetPropertyValue(Of Items)("ProductionItem", _productionItem, value)
            End Set
        End Property

        Public Property MustRun As Boolean?
            Get
                Return _mustRun
            End Get
            Set(value As Boolean?)
                SetPropertyValue(Of Boolean?)("MustRun", _mustRun, value)
            End Set
        End Property

        Public Property PriorityLevel As Integer?
            Get
                Return _priorityLevel
            End Get
            Set(value As Integer?)
                SetPropertyValue(Of Integer?)("PriorityLevel", _priorityLevel, value)
            End Set
        End Property

        Public Property Project As Project
            Get
                Return _project
            End Get
            Set(value As Project)
                SetPropertyValue(Of Project)("Project", _project, value)
            End Set
        End Property

        <Association("Project-ProductionSchedule")>
        Public Property ProjectDetail As ProjectDetails
            Get
                Return _projectDetail
            End Get
            Set(value As ProjectDetails)
                SetPropertyValue(Of ProjectDetails)("ProjectDetail", _projectDetail, value)
            End Set
        End Property

        <Size(SizeAttribute.Unlimited)>
        Public Property ScheduleNote As String
            Get
                Return _ScheduleNote
            End Get
            Set(value As String)
                SetPropertyValue(Of String)("ScheduleNote", _scheduleNote, value)
            End Set
        End Property

        Public Property Quantity As Double
            Get
                Return _Quantity
            End Get
            Set(value As Double)
                SetPropertyValue(Of Double)("Quantity", _Quantity, value)
            End Set
        End Property

        Public Property ScheduleByQuantity As Boolean
            Get
                Return _scheduleByQuantity
            End Get
            Set(value As Boolean)
                SetPropertyValue(Of Boolean)("ScheduleByQuantity", _scheduleByQuantity, value)
            End Set
        End Property

        Public ReadOnly Property ScheduleTitle As String
            Get
                If Project IsNot Nothing AndAlso ProductionItem IsNot Nothing Then
                    Return String.Format("{0} - {1}", Project.Project, ProductionItem.ItemCode)
                ElseIf ProductionItem IsNot Nothing Then
                    Return ProductionItem.ItemCode
                ElseIf Project IsNot Nothing Then
                    Return Project.Project
                Else
                    Return String.Empty
                End If
            End Get
        End Property

        Public Sub New()
            MyBase.New()
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
            ' Place here your initialization code.
        End Sub

        Public Shadows Class FieldsClass
            Inherits DevExpress.Xpo.PersistentBase.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property Oid() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Oid"))
                End Get
            End Property
            Public ReadOnly Property Finish As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Finish"))
                End Get
            End Property
            Public ReadOnly Property Start As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Start"))
                End Get
            End Property
            Public ReadOnly Property MachineLine As MachineLine.FieldsClass
                Get
                    Return New MachineLine.FieldsClass(GetNestedName("MachineLine"))
                End Get
            End Property
            Public ReadOnly Property People As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("People"))
                End Get
            End Property
            Public ReadOnly Property MachineOperator As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("MachineOperator"))
                End Get
            End Property
            Public ReadOnly Property ProductionItem As Items.FieldsClass
                Get
                    Return New Items.FieldsClass(GetNestedName("ProductionItem"))
                End Get
            End Property
            Public ReadOnly Property MustRun As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("MustRun"))
                End Get
            End Property
            Public ReadOnly Property PriorityLevel As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("PriorityLevel"))
                End Get
            End Property
            Public ReadOnly Property Project As Project.FieldsClass
                Get
                    Return New Project.FieldsClass(GetNestedName("Project"))
                End Get
            End Property
            Public ReadOnly Property ProjectDetail As ProjectDetails.FieldsClass
                Get
                    Return New ProjectDetails.FieldsClass(GetNestedName("ProjectDetail"))
                End Get
            End Property
            Public ReadOnly Property ScheduleNote As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ScheduleNote"))
                End Get
            End Property
            Public ReadOnly Property Quantity As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Quantity"))
                End Get
            End Property
            Public ReadOnly Property ScheduleByQuantity As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ScheduleByQuantity"))
                End Get
            End Property
        End Class

        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property
    End Class
End Namespace

File: DXDAL/ProductionSchedule.cs
using System;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace DXDAL
{
    namespace SPGData
    {
        public class ProductionSchedule : AuditableBase
        {
            private DateTime _finish;
            private DateTime _start;
            private MachineLine _machineLine;
            private double? _people;
            private double? _machineOperator;
            private Items _productionItem;
            private bool? _mustRun;
            private int? _priorityLevel;
            private Project _project;
            private ProjectDetails _projectDetail;
            private string _scheduleNote;
            private double _Quantity;
            private bool _scheduleByQuantity;

            public DateTime Finish
            {
                get => _finish;
                set => SetPropertyValue<DateTime>("Finish", ref _finish, value);
            }

            public DateTime Start
            {
                get => _start;
                set => SetPropertyValue<DateTime>("Start", ref _start, value);
            }

            public MachineLine MachineLine
            {
                get => _machineLine;
                set => SetPropertyValue("MachineLine", ref _machineLine, value);
            }

            public double? People
            {
                get => _people;
                set => SetPropertyValue("People", ref _people, value);
            }

            public double? MachineOperator
            {
                get => _machineOperator;
                set => SetPropertyValue("MachineOperator", ref _machineOperator, value);
            }

            public Items ProductionItem
            {
                get => _productionItem;
                set => SetPropertyValue("ProductionItem", ref _productionItem, value);
            }

            public bool? MustRun
            {
                get => _mustRun;
                set => SetPropertyValue("MustRun", ref _mustRun, value);
            }

            public int? PriorityLevel
            {
                get => _priorityLevel;
                set => SetPropertyValue("PriorityLevel", ref _priorityLevel, value);
            }

            public Project Project
            {
                get => _project;
                set => SetPropertyValue("Project", ref _project, value);
            }

            [Association("Project-ProductionSchedule")]
            public ProjectDetails ProjectDetail
            {
                get => _projectDetail;
                set => SetPropertyValue("ProjectDetail", ref _projectDetail, value);
            }

            [Size(SizeAttribute.Unlimited)]
            public string ScheduleNote
            {
                get => _scheduleNote;
                set => SetPropertyValue("ScheduleNote", ref _scheduleNote, value);
            }

            public double Quantity
            {
                get => _Quantity;
                set => SetPropertyValue("Quantity", ref _Quantity, value);
            }

            public bool ScheduleByQuantity
            {
                get => _scheduleByQuantity;
                set => SetPropertyValue("ScheduleByQuantity", ref _scheduleByQuantity, value);
            }

            public string ScheduleTitle
            {
                get
                {
                    if (Project != null && ProductionItem != null)
                        return string.Format("{0} - {1}", Project.project, ProductionItem.ItemCode);
                    else if (ProductionItem != null)
                        return ProductionItem.ItemCode;
                    else if (Project != null)
                        return Project.project;
                    else
                        return string.Empty;
                }
            }

            public ProductionSchedule() : base()
            {
            }

            public ProductionSchedule(Session session) : base(session)
            {
            }

            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty Oid => new OperandProperty(GetNestedName("Oid"));
                public OperandProperty Finish => new OperandProperty(GetNestedName("Finish"));
                public OperandProperty Start => new OperandProperty(GetNestedName("Start"));
                public MachineLine.FieldsClass MachineLine => new MachineLine.FieldsClass(GetNestedName("MachineLine"));
                public OperandProperty People => new OperandProperty(GetNestedName("People"));
                public OperandProperty MachineOperator => new OperandProperty(GetNestedName("MachineOperator"));
                public Items.FieldsClass ProductionItem => new Items.FieldsClass(GetNestedName("ProductionItem"));
                public OperandProperty MustRun => new OperandProperty(GetNestedName("MustRun"));
                public OperandProperty PriorityLevel => new OperandProperty(GetNestedName("PriorityLevel"));
                public Project.FieldsClass Project => new Project.FieldsClass(GetNestedName("Project"));
                public ProjectDetails.FieldsClass ProjectDetail => new ProjectDetails.FieldsClass(GetNestedName("ProjectDetail"));
                public OperandProperty ScheduleNote => new OperandProperty(GetNestedName("ScheduleNote"));
                public OperandProperty Quantity => new OperandProperty(GetNestedName("Quantity"));
                public OperandProperty ScheduleByQuantity => new OperandProperty(GetNestedName("ScheduleByQuantity"));
            }

            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
